spec/latest/core/core.jsonld

```diff
@@ -321,6 +322,15 @@
       "comment": "A runtime error, used to report information beyond the returned status code.",
       "vs:term_status": "testing"
     },
+    {
+      "@id": "hydra:errorCode",
+      "@type": "rdf:Property",
+      "subPropertyOf": "hydra:statusCode",
+      "label": "error code",
+      "comment": "The HTTP status code that is considered an error",
+      "domain": "hydra:Error",
+      "vs:term_status": "testing"
+    },

```

Comment:
Is this necessary? Even looking at the context itself, the property name defined RFC7807 is also simply `status`

Reply:
It was the only approach to have an `hydra:Error` in the result. As I wrote in the description of this pull request - I won't die for it. The only issue I have is that problem+json denotes a, well... problem, an error. The `hydra:Status` does not carry that semantics, `hydra:Error` does.

Reply:
Yes, IIRC it was somewhere proposed to actually drop `Error` in favour of `Status`, wasn't it?

Reply:
What's the correlation between `hydra:Error` and the response format that will adhere to RFC 7807? I guess it would be interesting for consumers of an API to know whether the response can be parsed according to RFC 7807 or not.

That will of course be possible to indicate with `Content-Type` as well, but knowing which responses to expect from a resource, including which are erroneous and parseable as RFC 7807 may be information that is useful to know before performing an HTTP request.

Reply:
Check the example in [index.html](https://github.com/HydraCG/Specifications/pull/218/files#diff-91d310ab3038fb1055c658d081cfac162ab60337751b95f239817a8660d0c69cR1277-R1289)

The response is an `application/problem+json`. So yes, parseable according to RFC7807.

If the spec was to state that a Hydra API `MUST` include the error context link in error responses then technically the clients can expect every error response to be a `hydra:Error` shaped as problem+json document.

What do you think about `MUST` @alien-mcl ?

Reply:
Currently there is no strong requirement to return either RDF based resource describing a `hydra:Error` or RFC7807 compatible JSON with context. I believe that in case an API is hydra driven, RFC7807 response could have the error context mandatory. I don't know whether we shall recommend RFC7807 over pure RDF response - how do you feel?

Reply:
The only thing that supports `hydra:Error` is the fact it matches RFC7807 semantics that is strictly bound to 4XX (and maybe 5XX). The `hydra:Status` resource could be in theory be attached to 3XX (I didn't find the RFC forbidding it - correct me if I'm wrong) or some of the 2XX (i.e. 202 Accepted).
Maybe `hydra:Status` and `hydra:Error` should be disjoined from each other - `hydra:Status` only for API documentation level to describe possible response and `hydra:Error` for 4XX and 5XX inline responses?

Reply:
> I don't know whether we shall recommend RFC7807 over pure RDF response

I think it makes sense to always return `application/problem+json` with a context link. This way even non-RDF clients or, more importantly, intermediaries will have a better way of understanding error responses.

Reply:
>I think it makes sense to always return `application/problem+json` with a context link. This way even non-RDF clients or, more importantly, intermediaries will have a better way of understanding error responses.

I agree a `application/problem+json` content type should at least be strongly encouraged, and perhaps even required, for `4xx` and `5xx` responses.

>The `hydra:Status` resource could be in theory attached to 3XX (I didn't find the RFC forbidding it - correct me if I'm wrong) 

Indeed, [`300 Multiple Choices`](https://tools.ietf.org/html/rfc7231#section-6.4.1) responses `SHOULD` contain a payload:

> For request methods other than HEAD, the server SHOULD generate a payload in the 300 response containing a list of representation metadata and URI reference(s) from which the user or user agent can choose the one most preferred.